### ISSUE TITLE
Use AsynchronousFileChannel for FileSource

### ISSUE THREAD

**Issue by bblfish** [akka/akka#30031]
I was a bit surprised reading the [Streaming File IO](https://doc.akka.io/docs/akka/current/stream/stream-io.html#streaming-file-io) docs to read 

> Please note that these operators are backed by Actors and by default are configured to run on a pre-configured threadpool-backed dispatcher dedicated for File IO. This is very important as it isolates the blocking file IO operations from the rest of the ActorSystem allowing each dispatcher to be utilised in the most efficient way. 

Looking around I found that [AsyncFileChannel](https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/nio/channels/AsynchronousFileChannel.html) was implemented differently on different Ones. So on Windows apparently it [does use underlying asynchronous IO](https://stackoverflow.com/questions/54213736/asynchronous-file-i-o-via-posix-aio-or-windows-overlapped-io-in-java). 

It would be interesting to find out on which other platforms this makes a difference.

**Comment by johanandren** [comment]
There was some previous investigation ending up with async filechannel being too much slower to be worth it. Discussions/investigations back in 2016 https://github.com/akka/akka/issues/21732

**Comment by bblfish** [comment]
It looks like those benchmarks did not specify what OS they were running on. It may be worth running them again on Windows and I guess Solaris where I the JVM can use the OS level async file features. Other OSes to check would be BSD and perhaps even MacOS with APFS running after checking if the Java implementations are using the OS level features as they seem to be on Windows. 

I checked if Rust did a better job, but they too have problems because they need different implementations per OS. So they don't have a good cross platform answer yet. But it won't take long for them to come to the conclusion that they may need to bite the bullet and make multiple implementations. 

One could see both working together: for OSes that support it, use the AsynFileChannel for file IO. For those that don't use what is currently there.

**Comment by johanandren** [comment]
I'd say Linux is the most important target given that is a very high percentage of Akka based systems will end up there, with Windows and MacOS on a second (but definitely not very interesting with remote filesystem as it is very unlikely how the stream reads there is the bottleneck). BSDs and Solaris is definitely low enough a priority that I wouldn't recommend spending time benchmarking those.

If you, or anyone else in the community, has time to and wants to run benchmarks comparing AsyncFileChannel using the PRs referenced in #21732 and report back if there are different results to those already listed in that ticket that'd surely be interesting.

**Comment by bblfish** [comment]
I agree. Even if Windows is not that important, it could be really useful to know what type of difference it makes there (or any OS that implements it correctly), if only to help push the Linux community to get their act together.
